internal/tasks/resource_scrape_test.go

```diff
@@ -76,33 +79,33 @@ func TestResourceScraping(baseT *testing.T) {
 
 		//first ScrapeNextResource() should scrape project1/foo
 		clock.StepBy(time.Hour)
-		t.Must(ExecuteOne(c.PollForResourceScrapes()))
+		t.Must(job.ProcessOne())
 		tr.DBChanges().AssertEqualf(`
-				INSERT INTO assets (id, resource_id, uuid, size, usage, next_scrape_at) VALUES (1, 1, 'asset1', 1000, '{"singular":400}', %[1]d);
-				INSERT INTO assets (id, resource_id, uuid, size, usage, next_scrape_at) VALUES (2, 1, 'asset2', 2000, '{"singular":1000}', %[1]d);
+				INSERT INTO assets (id, resource_id, uuid, size, usage, next_scrape_at, never_scraped) VALUES (1, 1, 'asset1', 0, '{"singular":0}', %[1]d, TRUE);
+				INSERT INTO assets (id, resource_id, uuid, size, usage, next_scrape_at, never_scraped) VALUES (2, 1, 'asset2', 0, '{"singular":0}', %[1]d, TRUE);
```

Comment:
I would not expect fixtures to change here

Reply:
There is a functional change there that I did not note in the PR description, in `internal/tasks/resource_scrape.go`, lines 169-191 on the red side. Upon discovering a new asset during resource scraping, we would previously do an asset scrape immediately and write the result in a single step. That's an optimization because we don't need to involve the asset-scrape loop right away, but a comparatively tiny one.

In the new implementation, I ran into trouble here because we should be counting the inline asset scrape into the metrics of the asset scrape, but we cannot do that since we are in a different job at this point (and the counter metrics are hidden inside the job instances and thus not accessible from this package). I decided to instead remove the micro-optimization and just have resource scraping create all assets without size and usage and with `NextScrapeAt = c.TimeNow()` to force an immediate asset scrape. Therefore, we see the DB fixture change to reflect that the asset was created, but not scraped yet.